sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_properties.asciidoc

```diff
@@ -0,0 +1,172 @@
+
+`buffer_location` property is defined as a part of `SYCL_INTEL_buffer_location` extension. Once passed in, buffer_location property is passed onto runtime libraries as well as the template arguments for annotated_ptr.
+
+== Proposal
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification, Section 6.3.3 "Feature test macros". Therefore, an
+implementation supporting this extension must predefine the macro
+`SYCL_EXT_ONEAPI_USM_PROPERTIES` to one of the values defined in the table below.
+Applications can test for the existence of this macro to determine if the
+implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features
+that the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value |Description
+|1     |Initial extension version
+|===
+
+
+=== New usm allocation template
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template <typename T, typename PropertyListT = property_list<>>
+annotated_ptr<T, PropertyListT> malloc_device(
+    size_t Count, const queue &Q, const sycl::ext::oneapi::experimental::properties &PropList = {},
+    const detail::code_location CodeLoc = detail::code_location::current()) {
+  annotated_ptr<T, PropertyListT> addr(malloc_device<T>(Count, Q.get_device(), Q.get_context(), PropList,
+                          CodeLoc));
+  return addr;
+}
+
+template <typename T, typename PropertyListT = property_list<>> // compile time properties
+annotated_ptr<T, PropertyListT> malloc_shared(
+    size_t Count, const queue &Q, const sycl::ext::oneapi::experimental::properties &PropList = {}, // runtime properties
+    const detail::code_location CodeLoc = detail::code_location::current()) {
+  ...
+}
+
+template <typename T, typename PropertyListT = property_list<>>
+annotated_ptr<T, PropertyListT> malloc_host(
+    size_t Count, const context &Ctxt, const sycl::ext::oneapi::experimental::properties &PropList = {},
+    const detail::code_location CodeLoc = detail::code_location::current()) {
+  ...
+}
+
+} // namespace sycl::ext::oneapi::experimental
+----
+
+The same setup is applied to other variant of the function signatures.
+
+Compile time properties can be pass into runtime properties within the allocation function.
+
+The table below describes the effects of associating each properties with each malloc function.
+
+|===
+|Property|Description
+
+|`buffer_location`
+|The `buffer_location` property adds the requirement that the memory must be
+ allocated to the correct device global memory location as defined in `SYCL_INTEL_buffer_location` extension. 
+ With `malloc_device` the returned device pointer must be in the target global memory.
+ with `malloc_shared`, compiler must implicitly migrate to the target device global memory.
+ with `malloc_host`, memory must be at the target device global memory when explicitly migrated.
```

Comment:
Maybe remove malloc_host, do nothing.

alloc on a host global mem id,

do not use device global memory terminology. "specified memory"